request_a_govuk_domain/request/models/application.py

```diff
@@ -23,6 +24,9 @@ class Application(models.Model):
 
     id = models.BigAutoField(primary_key=True)
     reference = models.CharField(max_length=REF_NUM_LENGTH, null=False)
+    owner = models.ForeignKey(User, on_delete=models.CASCADE, null=True)
+    when_submitted = models.DateTimeField(auto_now_add=True)
```

Comment:
I think `date_submitted`, or just `submitted` is clearer. To me `when_submitted` sounds like the name of an action like `on_delete`. Also maybe you can add a description field in the definition.